✨ feat(readme): Add 🎥 Codebase Navigation Video to README.md (#5243)
CircuitVerse/CircuitVerse: README.md

-----
[CircuitVerse](https://circuitverse.org) is a free, open-source platform which allows users to construct digital logic circuits online. We also offer the [Interactive Book](https://learn.circuitverse.org) which teaches users on the fundamentals of modern, digital circuits. Please also see our [documentation](https://docs.circuitverse.org) or [GitHub Wiki](https://github.com/CircuitVerse/CircuitVerse/wiki/).

> ## Getting Started with CircuitVerse 📹

Reviewer: ```suggestion
## Getting Started with CircuitVerse code base 📹
```

Author: done

To help new contributors get started, we've created a video guide titled **"Getting Started with CircuitVerse"**. This video covers:
- Introduction to CircuitVerse and its technical structure.
- Step-by-step setup using GitHub Codespaces.

> - A detailed walkthrough of the codebase directory structure and key files.

Reviewer: ```suggestion
- A high level walkthrough of the codebase directory structure and key files.
```

we haven't done detailed walk though

Author: done, also if needed would love doing a detailed version, would just need to know what it would constitute.

- Contribution guidelines for effective collaboration.

> 🎥 [Watch the "Getting Started with CircuitVerse" Video](https://youtu.be/TUdHWUdz_-8?si=g7rqFMfPS4CvisiF)  

Reviewer: ```suggestion
🎥 [Watch the "Getting Started with CircuitVerse" Video](https://youtu.be/TUdHWUdz_-8)  
```

Author: done
